examples/CampaignManagement/AddCompleteCampaignsUsingMutateJob.php

```diff
@@ -0,0 +1,486 @@
+
+    const NUMBER_OF_CAMPAIGNS_TO_ADD = 2;
+    const NUMBER_OF_ADGROUPS_TO_ADD = 2;
+    const NUMBER_OF_KEYWORDS_TO_ADD = 5;
+    const POLL_FREQUENCY_SECONDS = 1;
+    const MAX_TOTAL_POLL_INTERVAL_SECONDS = 60;
+
+    const PAGE_SIZE = 1000;
+
+    /** @var int the negative temporary ID used in mutate job operations. */
+    private static $temporaryId = 0;
+
+    public static function main()
+    {
+        // Either pass the required parameters for this example on the command line, or insert them
+        // into the constants above.
+        $options = (new ArgumentParser())->parseCommandArguments([
+            ArgumentNames::CUSTOMER_ID => GetOpt::REQUIRED_ARGUMENT,
+        ]);
+
+        // Generate a refreshable OAuth2 credential for authentication.
+        $oAuth2Credential = (new OAuth2TokenBuilder())->fromFile()->build();
+
+        // Construct a Google Ads client configured from a properties file and the
+        // OAuth2 credentials above.
+        $googleAdsClient = (new GoogleAdsClientBuilder())
+            ->fromFile()
+            ->withOAuth2Credential($oAuth2Credential)
+            ->build();
+        // We need to instantiate GoogleAdsFailure to make the library aware of
+        // possible failures during executing mutate jobs.
+        // This is specific to only this PHP client library.
+        $googleAdsFailure = new GoogleAdsFailure();
+
+        try {
+            self::runExample(
+                $googleAdsClient,
+                $options[ArgumentNames::CUSTOMER_ID] ?: self::CUSTOMER_ID
+            );
+        } catch (GoogleAdsException $googleAdsException) {
+            printf(
+                "Request with ID '%s' has failed.%sGoogle Ads failure details:%s",
+                $googleAdsException->getRequestId(),
+                PHP_EOL,
+                PHP_EOL
+            );
+            foreach ($googleAdsException->getGoogleAdsFailure()->getErrors() as $error) {
+                /** @var GoogleAdsError $error */
+                printf(
+                    "\t%s: %s%s",
+                    $error->getErrorCode()->getErrorCode(),
+                    $error->getMessage(),
+                    PHP_EOL
+                );
+            }
+        } catch (ApiException $apiException) {
+            printf(
+                "ApiException was thrown with message '%s'.%s",
+                $apiException->getMessage(),
+                PHP_EOL
+            );
+        }
+    }
+
+    /**
+     * Runs the example.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the client customer ID without hyphens
+     */
+    public static function runExample(GoogleAdsClient $googleAdsClient, $customerId)
+    {
+        $mutateJobServiceClient = $googleAdsClient->getMutateJobServiceClient();
+
+        // Creates a new mutate job.
+        $mutateJobResourceName =
+            $mutateJobServiceClient->createMutateJob($customerId)->getResourceName();
```

Comment:
In other examples we return the resource name directly rather than returning an object and then calling getResourceName externally, do you think it makes sense to change this for coherence?

Reply:
This is probably different situation.
In other cases, we didn't get "resource names", but get other attributes coming with the object (like getFeed()).
This case is like other examples such as AddCampaign, in which we just get a resource name returned directly from the server.